### ISSUE TITLE
UI - Execute SQL results

### ISSUE THREAD

**Issue by tlhackque** [sqlitebrowser/sqlitebrowser#1469]
Describe the new feature
--------------------------
The more I use "Execute SQL", the more I think the results of SELECT should be in a Browse Data object. - whether embedded in the window or in a separate tab.

The latest cases:

- Can't specify "encoding" (or "display format") (My Latin-1 data shows as "blob")
- Can't hide, filter (or rearrange) "SELECT * from"
- Right click in results table offers "use as filter", but that neither filters, generates SQL (e.g. add Where (...), nor updates the browse data tab.

Some (e.g. display format) change the SQL - and so perhaps don't seem to make sense with user-generated SQL. But, if (as suggested in #1463), the results became a temporary table in a data browser view, the power of the browser's UI could be applied.

This seems especially useful where Execute SQL is used to deal with situations that the browser UI can't express (e.g. join, relationships between columns (field1 > field 2), functions like max()...).

Having the data browser object be the single, consistent UI for viewing data would address these, and seem to be a good thing.  And you'd only have the one view's code to maintain.

Additional info
---------------
Please answer these questions before submitting your feature request.

#### Is your feature request related to an issue? Please include the issue number.
See also #1463

#### Does this feature exist in another product or project? Please provide a link.

Not sure.  Offhand, can't think of another tool that differentiates output from internal and user-supplied queries.  But this isn't my field of expertise.

#### Do you have a screenshot? Please add screenshots to help explain your idea.
No.

**Comment by chrisjlocke** [comment]
> Can't hide, filter (or rearrange) "SELECT * from"

By changing your select statement, you can select not only what columns you want, but the order in which they appear.  You can even have a column appear more than once.

**Comment by mgrojo** [comment]
The tables at Browse Data and Execute SQL should behave identically, as you are expecting. The only reason for not working is that the same approach cannot be used, since we had to parse and understand the select query. This same reason applies to #1177, as already commented by @MKleusberg. The proposed approach of making a temporary view may be easier. I don't know if it would have some price. I won't explore this option by now, since I have other pending implementations, but it seems a good idea.

Meanwhile, the "Use as Filter" option must not be visible in the Execute SQL table, since it is there useless. Maybe I fix that.

**Comment by tlhackque** [comment]
> By changing your select statement, you can select not only what columns you want, but the order in which they appear. You can even have a column appear more than once.

Yes, of course.  But that means typing each column name - which can be in the dozens.  That's why I used "SELECT \\\*\\" in .0

I'm always happier when software does busy work than when it makes me do it...
